src/Query.php

```diff
@@ -14,6 +14,58 @@ class Query
     protected $type;
     protected $wheres = [];
     protected $options = [];
+    protected $cacheQueryResults = false;
+    protected static $cachedRecords = [];
+
+    public static function eagerLoadAll(): Query
+    {
+        (new static)->cache()->get();
+
+        return new static;
+    }
+
+    public static function setRecordCache(Collection $records): Collection
+    {
+        return static::$cachedRecords = $records->keyBy('id');
+    }
+
+    public static function clearRecordCache(): Collection
+    {
+        return static::$cachedRecords = collect();
+    }
+
+    public static function addToRecordCache(Collection $records): Collection
+    {
+        $records = $records->keyBy('id');
+
+        static::setRecordCache(static::recordCache()->merge($records));
+
+        return $records;
+    }
+
+    public static function recordCache()
+    {
+        return collect(static::$cachedRecords);
+    }
+
+    public function cache(): Query
+    {
+        $this->cacheQueryResults = true;
+
+        return $this;
+    }
+
+    public function dontCache(): Query
+    {
+        $this->cacheQueryResults = false;
+
+        return $this;
+    }
+
+    public function shouldCache(): bool
```

Comment:
Should or Will?

Reply:
I've seen the context below now and `Should` makes sense

Reply:
This has been simplified so it is just the property name